quinn-h3/src/connection.rs

```diff
@@ -1,51 +1,63 @@
 use std::{
     collections::{BTreeMap, HashMap, VecDeque},
-    io::Cursor,
+    io::{self, Cursor},
     mem,
     pin::Pin,
     sync::{Arc, Mutex},
     task::{Context, Waker},
 };
 
-use bytes::{Bytes, BytesMut};
+use bytes::BytesMut;
 use futures::{AsyncRead, Future, Poll, Stream};
 use quinn::{IncomingBiStreams, IncomingUniStreams, RecvStream, SendStream};
 use quinn_proto::{Side, StreamId};
 
 use crate::{
-    frame::FrameStream,
+    frame::{self, FrameStream},
     proto::{
-        connection::{Connection, DecodeResult, Error as ProtoError, PendingStreamType},
+        self,
+        connection::{
+            Connection, DecodeResult, Error as ProtoError, Error as ConnectionError,
+            PendingStreamType,
+        },
         frame::{HeadersFrame, HttpFrame},
         ErrorCode, StreamType,
     },
     streams::{NewUni, RecvUni, SendUni},
     Error, Settings,
 };
 
+struct DriverError(Error, ErrorCode, String);
+
+macro_rules! driver_err {
```

Comment:
Why does this need to be a macro?

Reply:
It doesn't have to, I though it would bring call-site brevity.